src/patch.ml

```diff
@@ -36,27 +36,29 @@ let pp_hunk ~mine_no_nl ~their_no_nl ppf hunk =
     hunk.mine_start hunk.mine_len hunk.their_start hunk.their_len
     (unified_diff ~mine_no_nl ~their_no_nl hunk)
 
-let rec apply_hunk ~cleanly ~fuzz (last_matched_line, offset, lines) ({mine_start; mine_len; mine; their_start = _; their_len; their} as hunk) =
+let rec apply_hunk ~cleanly ~fuzz (last_matched_line, offset, rope) ({mine_start; mine_len; mine; their_start = _; their_len; their} as hunk) =
   let mine_start = mine_start + offset in
   let patch_match ~search_offset =
     let mine_start = mine_start + search_offset in
-    let rev_prefix, rest = Lib.List.rev_cut (Stdlib.max 0 (mine_start - 1)) lines in
-    let rev_actual_mine, suffix = Lib.List.rev_cut mine_len rest in
-    let actual_mine = List.rev rev_actual_mine in
-    if actual_mine <> (mine : string list) then
-      invalid_arg "unequal mine";
-    (* TODO: should we check their_len against List.length their? *)
+    let prefix = Rope.chop rope (Stdlib.max 0 (mine_start - 1)) in
+    let actual_mine = Rope.chop rope ~off:(Stdlib.max 0 (mine_start - 1)) mine_len in
+    let off = Stdlib.max 0 (mine_start - 1) + mine_len in
+    let suffix = Rope.shift rope off in
+    if Rope.to_strings actual_mine <> mine then
```

Comment:
instead of constructing a string list, it would be way more efficient to compare the other string list on the fly

Reply:
I'm not entirely sure what you propose here. We have a Rope.t and a string list -- we can:
- convert both to a string
- convert the string list to a rope
- convert the rope to a string list
- introduce a "check_if_rope_is_equal_to_string_list"

I guess you propose the last one, to avoid/minimize allocations?

Reply:
yes that was my thought exactly, thanks